tracing-subscriber/src/fmt/format/pretty.rs

```diff
@@ -0,0 +1,342 @@
+    pub fn with_source_location(mut self, display_location: bool) -> Self {
+        self.format = self.format.with_source_location(display_location);
+        self
+    }
+}
+
+impl<C, N, T> FormatEvent<C, N> for Format<Pretty, T>
+where
+    C: Collect + for<'a> LookupSpan<'a>,
+    N: for<'a> FormatFields<'a> + 'static,
+    T: FormatTime,
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, C, N>,
+        writer: &mut dyn fmt::Write,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        #[cfg(feature = "tracing-log")]
+        let normalized_meta = event.normalized_metadata();
+        #[cfg(feature = "tracing-log")]
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
+        #[cfg(not(feature = "tracing-log"))]
+        let meta = event.metadata();
+        write!(writer, "  ")?;
+        time::write(&self.timer, writer, true)?;
+
+        let style = if self.display_level {
+            Pretty::style_for(meta.level())
+        } else {
+            Style::new()
+        };
+
+        if self.display_target {
+            let bold = style.bold();
+            write!(
+                writer,
+                "{}{}{}: ",
+                bold.prefix(),
+                meta.target(),
+                bold.infix(style)
+            )?;
+        }
+        let mut v = PrettyVisitor::new(writer, true).with_style(style);
+        event.record(&mut v);
+        v.finish()?;
+        writer.write_char('\n')?;
+
+        let dimmed = Style::new().dimmed().italic();
+        let thread = self.display_thread_name || self.display_thread_id;
+        if let (true, Some(file), Some(line)) =
+            (self.format.display_location, meta.file(), meta.line())
+        {
+            write!(
+                writer,
+                "    {} {}:{}{}",
+                dimmed.paint("at"),
+                file,
+                line,
+                dimmed.paint(if thread { " " } else { "\n" })
+            )?;
+        } else if thread {
+            write!(writer, "    ")?;
+        }
+
+        if thread {
+            write!(writer, "{} ", dimmed.paint("on"))?;
+            let thread = std::thread::current();
+            if self.display_thread_name {
+                if let Some(name) = thread.name() {
+                    write!(writer, "{}", name)?;
+                    if self.display_thread_id {
+                        write!(writer, " ({:?})", thread.id())?;
+                    }
+                } else if !self.display_thread_id {
+                    write!(writer, " {:?}", thread.id())?;
+                }
+            } else if self.display_thread_id {
+                write!(writer, " {:?}", thread.id())?;
+            }
+            writer.write_char('\n')?;
+        }
+
+        let bold = Style::new().bold();
+        let span = event
+            .parent()
+            .and_then(|id| ctx.span(&id))
+            .or_else(|| ctx.lookup_current());
```

Comment:
I forget what happens here—do we get the span _regardless_ if it's active, and if `None` is returned, we get the currently active span? If so, can you add that as comment?

Reply:
`Event::parent` only returns `Some` if the event has an explicitly set parent (i.e., if one was passed to the macro). Otherwise we get it from the context. 

I can definitely add a comment. The other formatter implementations do the exact same behavior, though... it might be worth factoring this out somehow so that user-written event formatters don't have to reimplement it and it's provided by the API somehow. That seems like a follow-up change — there's a lot of room for API rework in 0.3 to try and make writing custom event formatters hurt a little less.

Reply:
sure, yeah—i'll make an issue about this to factor it and document it in some followups. i'm increasingly stateless about code that I worked on!